packages/medusa/src/loaders/plugins.ts

```diff
@@ -252,14 +251,15 @@ export async function registerServices(pluginDetails: PluginDetails, container:
       const loaded = require(fn).default
       const name = formatRegistrationName(fn)
 
-      if (!(loaded.prototype instanceof BaseService)) {
+      const isService = loaded.prototype instanceof BaseService || loaded.prototype instanceof TransactionBaseService
+      if (!isService) {
         const logger = container.resolve<Logger>("logger")
         const message = `Services must inherit from BaseService, please check ${fn}`
         logger.error(message)
         throw new Error(message)
       }
 
-      if (loaded.prototype instanceof PaymentService) {
+      if (loaded.prototype instanceof AbstractPaymentService) {
```

Comment:
thought: wouldn't it be best to make it possible for "legacy" payment providers to still work?

Reply:
And then remove the old interface in a coming release?

Reply:
Yes I can definitely do that, and depending on which one is used the payment provider service can use the appropriate method.

And i will decorate it as deprecated then